### PR TITLE
Check bbox

### PR DESCRIPTION
**Pull request by ocefpaf**
Only merge this one after #85. I rebased onto that to move forward as we wait for #85 to be merged.

### CODE REVIEW

File: pyoos/collectors/collector.py [hunk 1]
```diff
@@ -48,7 +51,7 @@ def get_bbox(self):
 
     def set_bbox(self, bbox):
         if bbox is not None:
-            if isinstance(bbox, list) or isinstance(bbox, tuple):
+            if isinstance(bbox, (list, tuple, np.array)) and len(bbox) == 4:
```
Comment (ocefpaf) [inline pyoos/collectors/collector.py]: Allow for numpy arrays and check for the right length. We can probably do better by just trying to make it a `tuple` and checking for `minx, miny, maxx, maxy`. Maybe I  should try that...

File: pyoos/collectors/collector.py [hunk 2]
```diff
@@ -47,12 +48,15 @@ def get_bbox(self):
...
-                bbox = tuple(bbox)
-            else:
-                raise ValueError("Not a recognized bbox. \
-                                  Must be in format: (minx, miny, maxx, maxy)")
+        bbox = tuple(bbox)
+        dx, dy = -1, -1
+        if len(bbox) == 4:
+            dx = bbox[2] - bbox[0]
+            dy = bbox[3] - bbox[1]
+        if not dx > 0 and not dy > 0:
```
Comment (ocefpaf) [inline pyoos/collectors/collector.py]: If I got this right we avoid users shooting themselves in the foot with a wrong `bbox` and provide a better info a bad bbox is provided.

Comment (Bobfrat) [reply pyoos/collectors/collector.py]: should this be an or? And just a suggestion, I find the logical NOTs harder to read than if you were to say 

```
if dx <= 0 and dy <= 0:
```

Comment (ocefpaf) [reply pyoos/collectors/collector.py]: `or` is better b/c it should fail if any of those evaluates and I agree about the logical NOT, thanks!

**Pull request merged**